src/rpc/methods/eth/filter/mod.rs

```diff
@@ -72,17 +72,41 @@ impl EthEventHandler {
 
             if let Some(filter_store) = &self.filter_store {
                 if filter_store.add(filter.clone()).is_err() {
-                    if let Some(tipset_filter_manager) = &self.tipset_filter_manager {
-                        let _ = tipset_filter_manager.remove(filter.id());
+                    if let Some(event_filter_manager) = &self.event_filter_manager {
+                        let _ = event_filter_manager.remove(filter.id());
                     }
-                    return Err(Error::msg("Removal error"));
+                    bail!("Adding filter failed.");
                 }
             }
             Ok(filter.id().clone())
         } else {
             Err(Error::msg("NotSupported"))
         }
     }
+
+    pub fn eth_new_block_filter(&self) -> Result<FilterID, Error> {
```

Comment:
Are we able to add some unit tests here?

Reply:
we have test for `tipset_filter_manager` methods used in here and rpc test for `eth_newBlockFilter` tests the same

Reply:
RPC test checks only the happy path. The logic here is not entirely trivial, so it would be good to have it covered by tests.

Reply:
added